src/main/java/Spike.java

```diff
@@ -108,34 +168,51 @@ public static Command isValidCommand(String command) {
         return null;
     }
 
+    /**
+     * Saves the latest task list into hard disk when user exits.
+     */
+    public static void saveChanges() {
+        try {
+            FileWriter fw = new FileWriter("data/Spike.txt");
+            String latestList = "";
+            for (Task task : taskList) {
+                latestList = latestList + task.toFileFormat() + "\n";
+            }
+            fw.write(latestList);
+            fw.close();
+        } catch (IOException e) {
+            printMsg("Oops, something went wrong with saving your file :(");
+        }
+    }
+
     /**
      * Adds task into the list and prints.
      */
-    public static void addTask(String command, String[] commandWords) throws DukeException {
-        switch (commandWords[0]) {
-        case "todo":
+    public static void addTask(Command c, String command, String[] commandWords) throws SpikeException {
+        switch (c) {
+        case TODO:
             if (command.length() <= 5) {
-                throw new DukeException("Hmmmm what to do? Think again?");
+                throw new SpikeException("Hmmmm what to do? Think again?");
             }
             ToDo newTD = new ToDo(command.substring(command.indexOf("todo") + 5));
             taskList.add(newTD);
             printAddedTask(newTD);
             break;
-        case "deadline":
+        case DEADLINE:
             // Extract description and deadline and pass to constructor
             if (commandWords.length <= 2 || command.indexOf("/by") == -1
```

Comment:
is it possible to seperate the comparisons so that it would be easier for others to understand

Reply:
Good idea, I will look into this.